### PR TITLE
Add shellcheck to pre-commit and make helper scripts clean

### PR DESCRIPTION
**Pull request by yarikoptic**
## What do these changes do?

Fix up found helper shell scripts so they become more robust and shellcheck clean.

Adds shellcheck to pre-commit

## Are there changes in behavior for the user?

no

## Checklist

- [ ] Fix remaining fails

<details>
<summary>I left following ones not fixed to see if CI would run pre-commit to ensure that nothing sneaks in
</summary> 

```shell
❯ pre-commit run --all shellcheck
ShellCheck v0.9.0........................................................Failed
- hook id: shellcheck
- exit code: 1

In tools/build-wheels.sh line 61:
for whl in ${ORIG_WHEEL_DIR}/*/${package_name}-*-linux_${arch}.whl; do
           ^---------------^ SC2231 (info): Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt .
                               ^-------------^ SC2231 (info): Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt .
                                                       ^-----^ SC2231 (info): Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt .


In tools/run_docker.sh line 35:
    docker run --rm -v $(pwd):/io "${manylinux1_image_prefix}${arch}" $dock_ext_args /io/tools/build-wheels.sh "$package_name"
                       ^----^ SC2046 (warning): Quote this to prevent word splitting.

For more information:
  https://www.shellcheck.net/wiki/SC2046 -- Quote this to prevent word splitt...
  https://www.shellcheck.net/wiki/SC2231 -- Quote expansions in this for loop...

```
</details>

### CODE REVIEW

**Comment by yarikoptic** [comment]
hm, it did fail but non-informatively

```
ShellCheck v0.9.0............................................................Failed
- hook id: shellcheck
- exit code: 1

Executable `docker` not found
```

apparently that pre-commit  action relies on docker :-/

```shell
❯ cat pre-commit-hooks.yaml.template
- id: shellcheck
  name: ShellCheck $VERSION
  description: Static analysis tool for shell scripts
  language: docker_image
  entry: $DOCKERHOST/$DOCKERHUB:$VERSION
  types: [shell]
❯ git grep DOCKER
autoupdate:export DOCKERHOST="docker.io"
autoupdate:export DOCKERHUB="koalaman/shellcheck"
autoupdate:  curl "https://registry.hub.docker.com/v2/repositories/$DOCKERHUB/tags" |
pre-commit-hooks.yaml.template:  entry: $DOCKERHOST/$DOCKERHUB:$VERSION
```

shellcheck seems written in haskell and has linux and OSX builds provided from github releases, but no windows...

anyways -- given above, I just want first to check if you want to keep it or should I drop it and just leave the fixes to scripts? (meanwhile I will push fixup for those remaining issues)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -185,4 +185,9 @@ repos:
       ^type of expression at "float number value" must be number
       but found type string$
 
+- repo: https://github.com/koalaman/shellcheck-precommit
+  rev: v0.9.0
+  hooks:
+  - id: shellcheck
```
Comment (webknjaz) [inline .pre-commit-config.yaml:191]: So I realized that I was using this other implementation, elsewhere…
```suggestion
- repo: https://github.com/shellcheck-py/shellcheck-py.git
  rev: v0.10.0.1
  hooks:
  - id: shellcheck
```
Let's go with it!

**Comment by webknjaz** [comment]
> anyways -- given above, I just want first to check if you want to keep it or should I drop it and just leave the fixes to scripts?

I've given it some thought and realized that I haven't run those scripts in years — we're standardizing around `cibuildwheel`. So it might make sense to just delete them from the repo. With that in mind, it's okay to keep the Python variant of shellcheck — it won't get triggered with no matching files until we add a new script, I suppose. Alternatively, we could just drop that as well.

**Comment by yarikoptic** [comment]
rright. Let me know if you want me to rebase, or feel welcome to just take what you want or just close -- up to you!

Cheers!

**Pull request closed without merging**